resources/server/sv_twitter.ts

```diff
@@ -316,12 +316,41 @@ onNet(events.TWITTER_GET_OR_CREATE_PROFILE, async () => {
 
   try {
     const profile = await getOrCreateProfile(identifier);
-    emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_SUCCESS, _source, profile);
+
+    // if we got null from getOrCreateProfile it means it doesn't exist and
+    // we failed to create it. In this case we pass the UI some default
+    // profile names it can choose from
+    if (!profile) {
+      const defaultProfileNames = await getDefaultProfileNames(identifier);
+      emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_NULL, _source, defaultProfileNames);
+    } else {
+      emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_SUCCESS, _source, profile);
+    }
   } catch (e) {
     twitterLogger.error(`Failed to get or create profile, ${e.message}`, {
       source: _source,
     });
-    emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_FAILURE, _source);
+    emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_FAILURE, _source, null);
```

Comment:
If there is a failure is there any reason we have to emit null? Can't it be assumed from a failure event?